### ISSUE TITLE
Precision Loss in JIT ScriptModule at float64 Tensor Init

### ISSUE THREAD

**Issue by gtyukasz** [pytorch/pytorch#115736]
### 🐛 Describe the bug

When initializing values in a float64 tensor either with a list or element-wise, they are truncated to float32 precision in a JIT `ScriptModule`.

**Example**
```python
import torch

class PrecisionLoss(torch.nn.Module):
    def forward(self, value: float):
        l = [value]
        t1 = torch.tensor(l, dtype=torch.float64)
        t2 = torch.empty([1], dtype=torch.float64)
        t2[0] = value
        print(f'{l} -> t1: {t1.item()}, t2: {t2.item()}')

module = PrecisionLoss()
script = torch.jit.script(module)

value = 1 + 2**-23
module.forward(value)
script.forward(value)

value = 1 + 2**-24
module.forward(value)
script.forward(value)
```
**Output**
```
[1.0000001192092896] -> t1: 1.0000001192092896, t2: 1.0000001192092896
[1.0000001192092896] -> t1: 1.0000001192092896, t2: 1.0000001192092896
[1.0000000596046448] -> t1: 1.0000000596046448, t2: 1.0000000596046448
[1.0000000596046448] -> t1: 1., t2: 1.
```
The value in the tensor is truncated in the script module, even though local scalar and list variables are full precision.

### Versions

```
PyTorch version: 2.1.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 14.2 (arm64)
GCC version: Could not collect
Clang version: 15.0.0 (clang-1500.1.0.2.5)
CMake version: Could not collect
Libc version: N/A

Python version: 3.11.6 (main, Oct  2 2023, 13:45:54) [Clang 15.0.0 (clang-1500.0.40.1)] (64-bit runtime)
Python platform: macOS-14.2-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Pro

Versions of relevant libraries:
[pip3] torch==2.1.1
[conda] Could not collect
```

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
